src/formats/XTC.cpp

```diff
@@ -63,6 +89,16 @@ void XTCFormat::set_positions(const std::vector<float>& x, Frame& frame) {
     }
 }
 
+void XTCFormat::get_positions(std::vector<float>& x, const Frame& frame) {
+    auto positions = frame.positions();
```

Comment:
Could you add an `assert(x.size() == 3 * positions.size())` here ? Just guarding against possible changes.

Reply:
Done, probably also good in `set_positions`.

Reply:
Yes !